File: software/examples/javascript/ExampleSimple.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletMoisture = require('Tinkerforge/BrickletMoisture');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'iB4';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var m = new BrickletMoisture(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Get current moisture value
        m.getMoistureValue(
            function(moisture) {
                console.log('Moisture: '+moisture);
            },
            function(error) {
                console.log('Error: '+error);
            }
        );
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_moisture.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Moisture m;
	moisture_create(&m, UID, &ipcon); 

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Get current moisture value
	uint16_t moisture;
	if(moisture_get_moisture_value(&m, &moisture) < 0) {
		fprintf(stderr, "Could not get value, probably timeout\n");
		exit(1);
	}

	printf("Moisture Value: %d\n", moisture);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
}
